src/main/java/edu/kis/powp/jobs2d/command/ICompoundCommand.java

```diff
@@ -1,12 +1,20 @@
 package edu.kis.powp.jobs2d.command;
 
+import edu.kis.powp.jobs2d.command.visitor.Visitor;
+
 import java.util.Iterator;
 
 /**
  * Interface extending Job2dDriverCommand to execute more than one command.
  */
 public interface ICompoundCommand extends DriverCommand {
 
-    public Iterator<DriverCommand> iterator();
+    Iterator<DriverCommand> iterator();
 
+    @Override
+    default void accept(Visitor visitor) {
+        while (iterator().hasNext()) {
+            iterator().next().accept(visitor);
```

Comment:
Not correct. The visitation of compound command should also call visit passing "this".